### ISSUE TITLE
layernorm fp16, The accuracy is still insufficient

### ISSUE THREAD

**Issue by CallmeZhangChenchen** [NVIDIA/TensorRT#4493]
# GPU: 4090
[06/18/2025-03:48:34] [I] TensorRT version: 10.10.0
[06/18/2025-03:48:34] [I] Loading standard plugins
[06/18/2025-03:48:34] [I] [TRT] [MemUsageChange] Init CUDA: CPU +2, GPU +0, now: CPU 26, GPU 390 (MiB)
[06/18/2025-03:48:37] [I] [TRT] [MemUsageChange] Init builder kernel library: CPU +1749, GPU +8, now: CPU 1977, GPU 398 (MiB)
[06/18/2025-03:48:37] [I] Start parsing network model.
[06/18/2025-03:48:37] [I] [TRT] ----------------------------------------------------------------
[06/18/2025-03:48:37] [I] [TRT] Input filename:   model.onnx
[06/18/2025-03:48:37] [I] [TRT] ONNX IR version:  0.0.8
[06/18/2025-03:48:37] [I] [TRT] Opset version:    17
[06/18/2025-03:48:37] [I] [TRT] Producer name:    pytorch
[06/18/2025-03:48:37] [I] [TRT] Producer version: 2.7.1
[06/18/2025-03:48:37] [I] [TRT] Domain:           
[06/18/2025-03:48:37] [I] [TRT] Model version:    0
[06/18/2025-03:48:37] [I] [TRT] Doc string:       
[06/18/2025-03:48:37] [I] [TRT] ----------------------------------------------------------------
[06/18/2025-03:48:37] [W] [TRT] ModelImporter.cpp:503: Make sure input input_ids has Int64 binding.
[06/18/2025-03:48:37] [W] [TRT] ModelImporter.cpp:503: Make sure input attention_mask has Int64 binding.
[06/18/2025-03:48:39] [I] Finished parsing network model. Parse time: 2.17323
[06/18/2025-03:48:39] [I] Set shape of input tensor input_ids for optimization profile 0 to: MIN=1x1 OPT=8x100 MAX=128x1024
[06/18/2025-03:48:39] [I] Set shape of input tensor attention_mask for optimization profile 0 to: MIN=1x1 OPT=8x100 MAX=128x1024
[06/18/2025-03:48:42] [W] [TRT] Detected layernorm nodes in FP16.
[06/18/2025-03:48:42] [W] [TRT] Running layernorm after self-attention with FP16 Reduce or Pow may cause overflow. Forcing Reduce or Pow
Layers in FP32 precision, or exporting the model to use INormalizationLayer (available with ONNX opset >= 17) can help preserving accur
acy.

# Reproduction steps

/usr/local/lib/python3.**/dist-packages/optimum/exporters/onnx/convert.py(467)
def export_pytorch()
Set opset=17

```
from sentence_transformers import (
    SentenceTransformer,
)
model = SentenceTransformer("Qwen/Qwen3-Embedding-0.6B", backend="onnx")
model.save_pretrained("export")
```

/usr/src/tensorrt/bin/trtexec --minShapes=input_ids:1x1,attention_mask:1x1 --optShapes=input_ids:8x100,attention_mask:8x100 --maxShapes=input_ids:128x1024,attention_mask:128x1024 --onnx=model.onnx --saveEngine=./model.plan --fp16

**Comment by pranavm-nvidia** [comment]
Can you elaborate on what the issue is? How large of a discrepancy are you seeing?

**Comment by CallmeZhangChenchen** [comment]
There is a warning when using trtexec to onnx.
the model to use INormalizationLayer (available with ONNX opset >= 17) can help preserving accuracy.

However, I found that for my model Opset version: 17, which contains INormalizationLayer, the accuracy is still very low
The following are the differences.

Forcing the INormalizationLayer to be fp32 can match the results of pytorch
... -0.0032405853271484375, 0.0171966552734375, 0.0084228515625, -0.00862884521484375, 0.0479736328125, 0.0261993408203125

fp16, there is a loss in accuracy, which is quite significant
... -0.0443115234375, -0.03564453125, -0.05865478515625, 0.0120086669921875, -0.0232086181640625, 0.04705810546875

**Comment by pranavm-nvidia** [comment]
Could you try checking whether this is a limitation of the model itself? You can convert the ONNX model to FP16 ([example](https://github.com/NVIDIA/TensorRT/tree/main/tools/Polygraphy/examples/cli/convert/04_converting_models_to_fp16#converting-onnx-models-to-fp16)) and check if you see a discrepancy with ONNX-RT.

**Comment by CallmeZhangChenchen** [comment]
Hi! @pranavm-nvidia I tested it and found that the fp16 after conversion did not have as much accuracy drop as tensorrt

polygraphy convert --fp-to-fp16 -o identityfp16.onnx model+opt17.onnx

```
import onnxruntime as ort
import numpy as np
 
session = ort.InferenceSession("identityfp16.onnx")

input1 = np.load('input_ids.npy')
input2 = np.load('attention_mask.npy')

input_names = session.get_inputs()
print("Input names:", [node.name for node in input_names])

inputs = {input_names[0].name: input1, input_names[1].name: input2}
outputs = session.run(None, inputs)

output_value = outputs[1] 
print("Output value:", output_value)
```
 outputs[1]: array([[ 0.01279657, -0.04986819, -0.00753332, ..., -0.00919925,
         0.04770797,  0.02603251]], dtype=float32)

**Comment by pranavm-nvidia** [comment]
It seems like the issue is actually on the export side. Even when I export the model with opset >= 17, I still don't see any `LayerNormalization`s in the resulting model - instead they are expanded into primitive ops. I also tried with `dynamo` export and see the same behavior. 

Maybe something like [AutoCast](https://nvidia.github.io/TensorRT-Model-Optimizer/guides/8_autocast.html) could help here, though ideally this could be fixed in the PyTorch ONNX exporter.

**Comment by CallmeZhangChenchen** [comment]
OK ，I'm merely giving feedback on this accuracy issue

I added it in trtexec --precisionConstraints=obey --layerPrecisions=**

The accuracy is still very high.
